lib/improver/tests/spotdata/spotdata/test_site_data.py

```diff
@@ -0,0 +1,300 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the spotdata.site_data"""
+
+
+import unittest
+import numpy as np
+import json
+from subprocess import call as Call
+from tempfile import mkdtemp
+from iris.tests import IrisTest
+
+from improver.spotdata.site_data import ImportSiteData as Plugin
+from improver.spotdata.site_data import set_utc_offset
+
+
+class Test_ImportSiteData(IrisTest):
+
+    """
+    Test the construction of SpotData site dictionaries from files and command
+    line definitions.
+
+    """
+
+    def setUp(self):
+        """Create components required for testing site_data."""
+
+        self.data_directory = mkdtemp()
+        self.site_data = [
+            {'latitude': -51.6927, 'bestdata_id': 16, 'altitude': 0,
+             'gmtoffset': -3.0, 'longitude': -57.8557},
+            {'latitude': -47.24417, 'bestdata_id': 15, 'altitude': 205,
+             'gmtoffset': -4.0, 'longitude': -72.585},
+            {'latitude': -40.35, 'bestdata_id': 10, 'altitude': 54,
+             'gmtoffset': 0.0, 'longitude': -9.8833},
+            {'latitude': -33.8, 'bestdata_id': 17, 'altitude': 25,
+             'gmtoffset': 2.0, 'longitude': 18.37},
+            {'latitude': -27.45, 'bestdata_id': 18, 'altitude': 38,
+             'gmtoffset': 10.0, 'longitude': 153.03}
+            ]
+
+        self.latitudes = [site['latitude'] for site in self.site_data]
+        self.longitudes = [site['longitude'] for site in self.site_data]
+        self.variables = ['latitude', 'longitude', 'altitude', 'utc_offset']
+
+    def tearDown(self):
+        """Remove temporary directories created for testing."""
+        Call(['rm', '-rf', self.data_directory])
+
+    @staticmethod
+    def change_key(list_of_dictionaries, oldkey, newkey):
+        """
+        Substitute a dictionary key in a list of dictionaries.
+
+        """
+        for dictionary in list_of_dictionaries:
+            dictionary[newkey] = dictionary.pop(oldkey)
+
+    def save_json(self, data):
+        """
+        Save the passed in data as a json file.
+
+        """
+        self.site_path = self.data_directory + '/site_data.json'
+        ff = open(self.site_path, 'w')
+        json.dump(data, ff, sort_keys=True, indent=4,
+                  separators=(',', ': ',))
+        ff.close()
+
+
```

Comment:
Just to note: all the tests below use the 
```
"""
Test things
"""
```
style of docstring, which I think we are moving away from in favour of:
```
"""Test things"""
```

This will be corrected by #89 in the next sprint anyway, so may not need to be changed right now.